Initial commit

smittytone opened the pull request.

ElectricImpSampleCode committed:
Initial ReadMe

ElectricImpSampleCode committed:
Update README.md

cat-haines committed:
Merge branch '1.0.0' of github.com:electricimp/Onewire into 1.0.0

cat-haines committed:
Renamed a couple properties and methods, changed inits return, and minor bug fix
- Renamed _ow --> _uart
- _Renamed slaves --> devices (down with colonialist language!!)
- _Renamed listSlaves --> discoverDevices (we're not listing, we're discovering)
  - discoverDevices returns the _devices array (seems to make sense)
- init now returns a bool (`if (success == 0)` looks like it's checking for failure)
  - init now only logs an error if _readErrorReason != 0
- READ_ERR_NO_DEVICS and READ_ERR_NO_BUS are 1 and 2 now (rather than strings)
  - This lets people check errors against named errors (much cleaner interface)
- Updated README with changes

cat-haines commented:
@smittytone - Looks good! 

I fixed whitespaces (indentations should be 4 spaces, not 2 or tabs) and updated some language (we're trying to stay away from master/slave terminology), and a couple other minor things.. let me know if you're comfortable with the changes and we can merge this to master.

smittytone commented:
Darn it - I was sure you’d said 2 space in the style doc. Noted for next time.

> On 1 Jul 2015, at 19:56, Matt Haines notifications@github.com wrote:
> 
> @smittytone https://github.com/smittytone - Looks good!
> 
> I fixed whitespaces (indentations should be 4 spaces, not 2 or tabs) and updated some language (we're trying to stay away from master/slave terminology), and a couple other minor things.. let me know if you're comfortable with the changes and we can merge this to master.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/electricimp/Onewire/pull/1#issuecomment-117795276.

cat-haines committed:
Added #require statement to constructor example

ElectricImpSampleCode committed:
Update Onewire.class.nut

smittytone commented:
Looks good - the changes are definite improvements.

I’ve tweaked the error reporting code, so that in place of an error code, it returns the code and the message in a table - more useful, I think. Updated the read me with your changes.

Have a look, and if you’re OK with it, I’ll merge.

> On 1 Jul 2015, at 19:56, Matt Haines notifications@github.com wrote:
> 
> @smittytone https://github.com/smittytone - Looks good!
> 
> I fixed whitespaces (indentations should be 4 spaces, not 2 or tabs) and updated some language (we're trying to stay away from master/slave terminology), and a couple other minor things.. let me know if you're comfortable with the changes and we can merge this to master.
> 
> —
> Reply to this email directly or view it on GitHub https://github.com/electricimp/Onewire/pull/1#issuecomment-117795276.

ElectricImpSampleCode committed:
Update Onewire.class.nut

cat-haines committed:
Added inforamtion about the error object
